pkg/util/cgroups/cgroupv1.go

```diff
@@ -7,33 +7,46 @@
 
 package cgroups
 
-import "path/filepath"
+import (
+	"path/filepath"
+)
 
 type cgroupV1 struct {
-	identifier  string
-	mountPoints map[string]string
-	path        string
-	fr          fileReader
-	pidMapper   pidMapper
+	identifier     string
+	mountPoints    map[string]string
+	path           string
+	fr             fileReader
+	pidMapper      pidMapper
+	inode          uint64
+	baseController string
 }
 
-func newCgroupV1(identifier, path string, mountPoints map[string]string, pidMapper pidMapper) *cgroupV1 {
+func newCgroupV1(identifier, path, baseController string, mountPoints map[string]string, pidMapper pidMapper) *cgroupV1 {
 	return &cgroupV1{
-		identifier:  identifier,
-		mountPoints: mountPoints,
-		path:        path,
-		pidMapper:   pidMapper,
-		fr:          defaultFileReader,
+		identifier:     identifier,
+		mountPoints:    mountPoints,
+		path:           path,
+		pidMapper:      pidMapper,
+		fr:             defaultFileReader,
+		baseController: baseController,
 	}
 }
 
 func (c *cgroupV1) Identifier() string {
 	return c.identifier
 }
 
+func (c *cgroupV1) Inode() uint64 {
+	if c.inode > 2 {
```

Comment:
I wanted to retry in case an error could be recoverable. However it doesn't matter because we're building new `cgroup{}` objects every 2 seconds anyway so I'll be applying the suggestion.

I have a question regarding the thread safety though. I don't see anywhere mentioned that this object should be thread safe and so far I think cgroup objects are parsed behind a mutex. Should we mention that these objects should be thread safe somewhere ?

Reply:
You're right that it's not mentioned, the map is being a mutex but the `Cgroup` object is not and is normally thread safe even though it's not documented.

Reply:
That's fair. It should be fixed now 👍